crates/wit/src/commands/pull.rs

```diff
@@ -0,0 +1,273 @@
+                        version: Some(release.version.clone()),
+                    };
+                    terminal.status("Downloaded", format!("release {release_pkg}"))?;
+
+                    let version = root_pkg.name.version.clone();
+                    if let Some(wit_version) = &version {
+                        let release_version = &release.version;
+                        if wit_version != release_version {
+                            terminal.warn(format!("Release version {release_version} doesn't match WIT package version {wit_version}"))?;
+                        }
+                    }
+
+                    for (package_id, package) in &decoded.resolve().packages {
+                        let name = &package.name;
+                        if !existing_pkgs.insert(name.clone()) {
+                            continue;
+                        }
+                        let path = self.write_package(decoded.resolve(), package_id)?;
+                        terminal.status(
+                            "Wrote",
+                            format!("package {name} to '{path}'", path = path.display()),
+                        )?;
+                    }
+                }
+                None => {
+                    terminal.warn(format!("No package found for {pkg}"))?;
+                }
+            }
+        }
+        Ok(())
+    }
+
+    fn parse_packages_state(&self) -> Result<(BTreeSet<PackageName>, BTreeSet<PackageName>)> {
+        let mut resolve = Resolve::new();
+
+        // If Resolve::push_dir succeeds there are no unresolved deps
+        match resolve.push_dir(&self.wit_dir) {
+            Ok(_) => {
+                return Ok((
+                    resolve.package_names.into_keys().collect(),
+                    Default::default(),
+                ))
+            }
+            Err(err) => log::debug!("Couldn't resolve packages: {err:#}"),
+        }
+
+        // Approximate the Resolve::push_dir process
+        // TODO: Try to expose this logic from wit-parser instead
+        let mut pkgs = match UnresolvedPackage::parse_dir(&self.wit_dir) {
+            Ok(pkg) => BTreeSet::from([pkg.name]),
+            Err(err) => {
+                log::debug!("Couldn't parse root package: {err:?}");
+                Default::default()
+            }
+        };
+        let mut deps = BTreeSet::new();
+        let deps_dir = self.wit_dir.join("deps");
+        if deps_dir.is_dir() {
+            for entry in deps_dir.read_dir()? {
+                let entry = entry?;
+                let path = entry.path();
+                let pkg = if entry.file_type()?.is_dir() {
+                    UnresolvedPackage::parse_dir(&path)
+                } else if path.extension().unwrap_or_default() == "wit" {
+                    UnresolvedPackage::parse_file(&path)
+                } else {
+                    continue;
+                }
+                .with_context(|| format!("Error parsing {path:?}"))?;
+
```

Comment:
Can we debug log here both `pkg.name` and `pkg.foreign_deps.keys()`?

Reply:
c63c145